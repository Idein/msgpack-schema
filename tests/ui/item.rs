use msgpack_schema::*;

mod serialize {
    use super::*;

    #[derive(Serialize)]
    #[tag = 1]
    struct S1 {
        x: String,
    }

    #[derive(Serialize)]
    #[optional]
    struct S2 {
        x: String,
    }

    #[derive(Serialize)]
    #[flatten]
    struct S5 {
        x: String,
    }

    #[derive(Serialize)]
    #[untagged]
    struct S3(String);

    #[derive(Serialize)]
    #[untagged]
    struct S4;
}

mod deserialize {
    use super::*;

    #[derive(Deserialize)]
    #[tag = 1]
    struct S1 {
        x: String,
    }

    #[derive(Deserialize)]
    #[optional]
    struct S2 {
        x: String,
    }

    #[derive(Deserialize)]
    #[flatten]
    struct S5 {
        x: String,
    }

    #[derive(Deserialize)]
    #[untagged]
    struct S3(String);

    #[derive(Deserialize)]
    #[untagged]
    struct S4;
}

fn main() {}
