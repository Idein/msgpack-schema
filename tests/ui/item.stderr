error: #[tag] at an invalid position
 --> $DIR/item.rs:7:5
  |
7 |     #[tag = 1]
  |     ^^^^^^^^^^

error: #[optional] at an invalid position
  --> $DIR/item.rs:13:5
   |
13 |     #[optional]
   |     ^^^^^^^^^^^

error: #[flatten] at an invalid position
  --> $DIR/item.rs:19:5
   |
19 |     #[flatten]
   |     ^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/item.rs:25:5
   |
25 |     #[untagged]
   |     ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/item.rs:29:5
   |
29 |     #[untagged]
   |     ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/item.rs:33:5
   |
33 |     #[untagged]
   |     ^^^^^^^^^^^

error: #[tag] at an invalid position
  --> $DIR/item.rs:41:5
   |
41 |     #[tag = 1]
   |     ^^^^^^^^^^

error: #[optional] at an invalid position
  --> $DIR/item.rs:47:5
   |
47 |     #[optional]
   |     ^^^^^^^^^^^

error: #[flatten] at an invalid position
  --> $DIR/item.rs:53:5
   |
53 |     #[flatten]
   |     ^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/item.rs:59:5
   |
59 |     #[untagged]
   |     ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/item.rs:63:5
   |
63 |     #[untagged]
   |     ^^^^^^^^^^^

error: #[untagged] at an invalid position
  --> $DIR/item.rs:67:5
   |
67 |     #[untagged]
   |     ^^^^^^^^^^^
